test/custom-parser.test.js

```diff
@@ -0,0 +1,194 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const request = require('request')
+const Fastify = require('..')
+const jsonParser = require('body/json')
+
+test('contentTypeParser method should exist', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  t.ok(fastify.contentTypeParser)
+})
+
+test('contentTypeParser should add a custom parser', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.post('/', (req, reply) => {
+    reply.send(req.body)
+  })
+
+  fastify.contentTypeParser.add('application/jsoff', function (req, done) {
+    jsonParser(req, function (err, body) {
+      if (err) return done(err)
+      done(body)
+    })
+  })
```

Comment:
I think we should should wrap `contentTypeParser.add` into a single function, and hide the `contentTypeParser` property.